### PR TITLE
Sync ROS2 branch with the master branch

### PR DESCRIPTION
**Pull request by JafarAbdi**
This doesn't include the new python API.

To test run the demos

### CODE REVIEW

**Comment by JafarAbdi** [comment]
`Run asarium/clang-tidy-action@v1` is failing with `Error: ENOENT: no such file or directory, open '/opt/ros/rolling/include/rclcpp/rclcpp/experimental/intra_process_manager.hpp'` :/, I have no idea what that means, the file do exists in rolling and humble

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -8,26 +8,31 @@ on:
...
     strategy:
       fail-fast: false
       matrix:
         env:
+          - IMAGE: rolling-source
+            TARGET_CMAKE_ARGS: >-
+              -DCMAKE_BUILD_TYPE=Release
           - IMAGE: humble-source
-            CXXFLAGS: >-
-             -Werror -Wall -Wextra -Wwrite-strings -Wunreachable-code -Wpointer-arith
```
Comment (JafarAbdi) [inline .github/workflows/ci.yaml:19]: I removed these since they're included in the `moveit_package` cmake macro

**Comment by rhaschke** [comment]
I will have a look into this over the next weekend - hopefully.

**Comment by rhaschke** [comment]
Closing in favour of #416

**Pull request closed without merging**